solutions/usaco-738.mdx

```diff
@@ -2,19 +2,114 @@
 id: usaco-738
 source: USACO Silver 2017 US Open
 title: Paired Up
-author: Sofia Yang
+author: Sofia Yang, Ryan Chou
 
 ---
 
-[Official Analysis](http://www.usaco.org/current/data/sol_pairup_silver_open17.html)
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_pairup_silver_open17.html)
 
+In the sample input, we have the following cows:
+$8, 5, 5, 2$, and there's two ways to pair these up.
+<ol>
+<li>$(8, 5), (5, 2)$</li>
+<li>$(8, 2), (5, 5)$</li>
+</ol>
+
+The first case would take $13$ time units to finish milking all cows.
+
+The second case would take $10$ time units to finish milking all cows.
+
+This test case consists of 3 distinct values: $8, 5, 2$, let's call them $A, B, C$. These would follow the paradigm $A < B < C$.
+
+Therefore, $A+C < B + C$.
+
+We should always pair up the greatest value ($C$), with the smallest value ($A$), in order to achieve the most optimal milking time.
+
+## Implementation
 **Time Complexity:** $\mathcal{O}(N\log N)$
 
 (Because it requires sorting the input, which is $\mathcal{O}(N\log N)$). 
 
 
 <LanguageSection>
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <fstream>
+#include <cmath>
+#include <algorithm>
+#include <vector>
+using namespace std;
+
+typedef pair<int,int> pii;
+vector<pii> V;
+int N;
```

Comment:
why not make these local variables

Reply:
i don't think this really makes a difference? (added it anyway)

Reply:
it doesn't, globals are just bad practice in general

Reply:
oh ok, got it :)